int32[] set_index
int32[] point_index